File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.ControlBuilders/Views/ArticleView.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI

Public Class ArticleView
	Inherits CMSPartBase
	Public Property Category() As String

	Protected Overrides Sub Render(ByVal writer As HtmlTextWriter)
		writer.Write("<h1>" & Title & "</h1>")
		writer.Write("<p>This is a list of " & PageSize.ToString() & " articles in " & Category & ".</p>")
	End Sub
End Class

File: aspnet-in-practice/06/CS/CustomControls.ControlBuilders/Views/ArticleView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;

namespace CustomControls.ControlBuilders
{
	public class ArticleView : CMSPartBase
	{
		public string Category { get; set; }
		protected override void Render(HtmlTextWriter writer)
		{
			writer.Write("<h1>" + Title + "</h1>");
			writer.Write("<p>This is a list of " + PageSize.ToString() + " articles in " + Category + ".</p>");
		}
	}
}